clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -402,26 +403,38 @@ public void refreshCommittedOffsetsIfNeeded() {
         }
     }
 
-    @Override
-    public void close() {
+    public void close(long timeoutMs, long requestTimeoutMs) {
         // we do not need to re-enable wakeups since we are closing already
         client.disableWakeups();
+        long startTimeMs = time.milliseconds();
         try {
-            maybeAutoCommitOffsetsSync();
+            timeoutMs = Math.min(timeoutMs, requestTimeoutMs);
+            maybeAutoCommitOffsetsSync(timeoutMs);
+        } finally {
+            super.close();
```

Comment:
`super.close()` queues a `LeaveGroup` request for sending to coordinator. Any pending commits have already been queued before the `LeaveGroup`. The following loop sends and awaits response from all queued requests. Since they are sent in order, `LeaveGroup` will be processed at the end and hence the commits shouldn't be impacted by the `LeaveGroup`. I have added a comment to the code. Does this make sense?